src/parser/statement.js

```diff
@@ -491,12 +491,13 @@ export default class StatementParser extends ExpressionParser {
     if (this.match(tt._catch)) {
       const clause = this.startNode();
       this.next();
-
-      this.expect(tt.parenL);
-      clause.param = this.parseBindingAtom();
-      this.checkLVal(clause.param, true, Object.create(null), "catch clause");
-      this.expect(tt.parenR);
-
+      if (this.match(tt.parenL) || !this.hasPlugin("optionalCatchBinding")) {
+        this.expect(tt.parenL);
+        clause.param = this.parseBindingAtom();
+        this.checkLVal(clause.param, true, Object.create(null), "catch clause");
+        this.expect(tt.parenR);
+      }
+      clause.param = null;
```

Comment:
Hmmm.  This is causing a failure in one of my tests.  I'll have to look at more.